### PR TITLE
Austin Carr-Jones

### PR DESCRIPTION
**Pull request by austincarrjones**

### CODE REVIEW

File: lib/item.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Item
+
+  attr_reader :name,
+              :price
+
+  def initialize(info_hash)
+    @info_hash = info_hash
```
Comment (KatBrandt) [inline lib/item.rb:7]: We don't need to store the whole info_hash in an attribute.

File: lib/item.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    @name = info_hash[:name]
+    @price = info_hash[:price].gsub('$', '').to_f
```
Comment (KatBrandt) [inline lib/item.rb:9]: Yes! Great job recognizing that you need to change this string to a float. Nice work here.

File: lib/food_truck.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @name = name
+    @inventory = {}
+  end
+  
+  def stock(item, stock)
+    if @inventory.key?(item)
+      @inventory[item] += stock
+    else
+      @inventory[item] = stock
+    end
```
Comment (KatBrandt) [inline lib/food_truck.rb:16]: Very nice job with this method!

File: lib/food_truck.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  end
+  
+  def check_stock(item)
+    if @inventory.include?(item)
+      @inventory[item]
+    else
+      0
+    end
```
Comment (KatBrandt) [inline lib/food_truck.rb:24]: Yup, exactly, nice work.

File: lib/event.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  def add_food_truck(food_truck)
+    @food_trucks << food_truck
+  end
+
+  def food_truck_names
+    names_array = []
+    @food_trucks.each do |truck|
+      names_array << truck.name
+    end
+    names_array
```
Comment (KatBrandt) [inline lib/event.rb:20]: Yup, this works! Can you also see how to do this with map?

File: lib/event.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  def food_trucks_that_sell(item)
+    trucks_array = []
+    @food_trucks.each do |truck|
+      if truck.inventory.include?(item)
+        trucks_array << truck
+        #binding.pry
+      end
+    end
+    trucks_array
```
Comment (KatBrandt) [inline lib/event.rb:31]: Yup! This works. Nice job. Can you rewrite this using find_all?

File: spec/food_truck_spec.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  it 'can stock more of the same item and check stock' do
+    food_truck = FoodTruck.new("Rocky Mountain Pies")
+    item1 = Item.new({name: 'Peach Pie (Slice)', price: "$3.75"})
+    food_truck.stock(item1, 30)
+    expect(food_truck.check_stock(item1)).to eq(30)
+    
+    food_truck.stock(item1, 25)
+    expect(food_truck.check_stock(item1)).to eq(55)
```
Comment (KatBrandt) [inline spec/food_truck_spec.rb:34]: Great job with this testing!!! Now that you've stocked the same item twice AND stocked a second item and everything passes, we can feel confident that your method is working for all situations. This is a nice, thorough test.

File: spec/event_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    food_truck3 = FoodTruck.new("Palisade Peach Shack") 
+    food_truck3.stock(item1, 65) 
+    event.add_food_truck(food_truck1)   
+    event.add_food_truck(food_truck2) 
+    event.add_food_truck(food_truck3)
+
+    expect(event.food_trucks).to eq([food_truck1, food_truck2, food_truck3])
+    expect(event.food_truck_names).to eq(["Rocky Mountain Pies", "Ba-Nom-a-Nom", "Palisade Peach Shack"])
+    expect(event.food_trucks_that_sell(item1)).to eq([food_truck1, food_truck3])
+    expect(event.food_trucks_that_sell(item4)).to eq([food_truck2])
```
Comment (KatBrandt) [inline spec/event_spec.rb:33]: Testing kind of fell apart here. :) Make sure to write one test for each method we write. Also think about what setup is needed for that particular test. When we're testing that we can add food trucks to an event, we don't need to stock the truck with items, we just need to make sure we can add the truck to the event.

**Review by KatBrandt — COMMENTED** [review]
Your code and tests look great Austin. I recommending working through the rest of this, getting through Iteration 3 and making sure everything makes sense, then practicing it again! :) You got this.

File: spec/event_spec.rb (hunk 8 above)
Comment (austincarrjones) [reply spec/event_spec.rb:33]: Yes, I started rushing at the end and figured I'd try and focus on testing code versus writing out the it blocks. Your video (still going through) is super helpful and I'll stick to writing multiple & separate tests moving forward 😁

**Pull request closed without merging**